src/addon-manager-proxy.ts

```diff
@@ -161,23 +222,24 @@ class AddonManagerProxy extends EventEmitter {
    * @method onMsg
    * Called whenever a message is received from the gateway.
    */
-  onMsg(msg) {
-    this.verbose && console.log('AddonManagerProxy: Rcvd:', msg);
+  onMsg(genericMsg: Message2): void {
+    this.verbose && console.log('AddonManagerProxy: Rcvd:', genericMsg);
 
-    switch (msg.messageType) {
+    switch (genericMsg.messageType) {
       case MessageType.PLUGIN_UNLOAD_REQUEST:
         this.unloadPlugin();
         return;
 
       case MessageType.API_HANDLER_UNLOAD_REQUEST: {
+        const msg = <APIHandlerAPIRequest>genericMsg;
```

Comment:
Shouldn't this be using `APIHandlerUnloadRequest`?

Reply:
Indeed :grimacing:

Reply:
Done